services/provider/client/client.go

```diff
@@ -142,7 +142,15 @@ const (
 	StorageTypeSharedfilesystem
 )
 
-func (cc *OCSProviderClient) FulfillStorageClassClaim(ctx context.Context, consumerUUID, storageClassClaimName, encryptionMethod string, storageType StorageType) (*pb.FulfillStorageClassClaimResponse, error) {
+func (cc *OCSProviderClient) FulfillStorageClassClaim(
+	ctx context.Context,
+	consumerUUID,
+	storageClassClaimName,
+	encryptionMethod,
+	storageProfile string,
+	storageType StorageType,
```

Comment:
above will not work storageClassClaimName is string not StorageType and it will mix multiple types. if you will need to see it in order we need to change as below

```
        ctx context.Context,
	consumerUUID,
	storageClassClaimName string,
	storageType StorageType
	storageProfile,
	encryptionMethod string,
```

Reply:
I just listed the order, I didn't try to give you the actual code suggestion. 
And of course, we should have a type definition for each arg. 
In my opinion, the type grouping feature just reduces the readability of the code and facilitates the making of wrong decisions just to keep the code a little bit more "clean"